src/convertToObject.js

```diff
@@ -6,7 +6,20 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const stylesObject = sourceString
+    .split(';')
+    .map((styles) => styles.trim())
+    .filter((styles) => styles.length > 0)
+    .map((styles) => styles.split(':'))
+    .reduce(
+      (object, styles) => ({
+        ...object,
+        [styles[0].trim()]: styles[1].trim(),
```

Comment:
The use of the spread operator inside the reduce callback is not recommended for building up new objects in each iteration due to performance concerns. It's better to directly assign the new property to the accumulator object.